sculture-rest/src/main/java/sculture/models/response/BaseStoryResponse.java

```diff
@@ -40,110 +33,60 @@ public void setId(long id) {
     private long report_count;
 
 
-    public BaseStoryResponse(Story story, List<String> tags, String owner_username, String editor_username) {
+    public BaseStoryResponse(Story story, TagStoryDao tagStoryDao, UserDao userDao) {
         this.id = story.getStory_id();
         this.title = story.getTitle();
         this.creation_date = story.getCreate_date();
         this.update_date = story.getLast_edit_date();
         this.last_editor.id = story.getLast_editor_id();
-        this.last_editor.username = editor_username;
         this.owner.id = story.getOwner_id();
-        this.owner.username = owner_username;
-        this.tags = tags;
         this.positive_vote = story.getPositive_vote();
         this.negative_vote = story.getNegative_vote();
         this.report_count = story.getReport_count();
-    }
-    public BaseStoryResponse(Story story)
-    {
-        this.id = story.getStory_id();
-        this.title = story.getTitle();
-        this.creation_date = story.getCreate_date();
-        this.update_date = story.getLast_edit_date();
-        this.last_editor.id = story.getLast_editor_id();
-        this.owner.id = story.getOwner_id();
-        this.positive_vote = story.getPositive_vote();
-        this.negative_vote = story.getNegative_vote();
-        this.report_count = story.getReport_count();
-    }
-
-    public long getPositive_vote() {
-        return positive_vote;
-    }
-
-    public void setPositive_vote(long positive_vote) {
-        this.positive_vote = positive_vote;
-    }
 
-    public long getNegative_vote() {
-        return negative_vote;
-    }
+        this.tags = tagStoryDao.getTagTitlesByStoryId(this.id);
 
-    public void setNegative_vote(long negative_vote) {
-        this.negative_vote = negative_vote;
-    }
-
-    public long getReport_count() {
-        return report_count;
-    }
-
-    public void setReport_count(long report_count) {
-        this.report_count = report_count;
+        this.owner.username = userDao.getById(story.getOwner_id()).getUsername();
+        this.last_editor.username = userDao.getById(story.getLast_editor_id()).getUsername();
     }
 
     public long getId() {
         return id;
     }
 
-    public void setId(long id) {
```

Comment:
I have removed all setters. There is no way to modify this classes other than constructor.